chain/lf3/f3.go

```diff
@@ -203,3 +204,21 @@ func (fff *F3) IsRunning() bool {
 func (fff *F3) Progress() gpbft.Instant {
 	return fff.inner.Progress()
 }
+
+func (fff *F3) ListParticipants() ([]address.Address, error) {
+	currentManifest := fff.inner.Manifest()
+	if currentManifest == nil {
+		return nil, nil
+	}
+	progress := fff.inner.Progress()
+	participantIDs := fff.leaser.getParticipantsByInstance(currentManifest.NetworkName, progress.ID)
```

Comment:
I'd consider adding a method on the leaser itself given that the leaser needs to internally get the progress/manifest. It would also be nice to return the actual leases (and, e.g., return not-yet-active leases?).

Reply:
By actual leases, I mean the start/end instance. I don't think we need the rest of the information.

Reply:
How about [this](https://github.com/filecoin-project/lotus/pull/12608/commits/b2266e59f7148319196fc302a24a0a49d27b2708)?